rack-attack.gemspec

```diff
@@ -32,6 +32,7 @@ Gem::Specification.new do |s|
   s.add_development_dependency "bundler", ">= 1.17", "< 3.0"
   s.add_development_dependency 'minitest', "~> 5.11"
   s.add_development_dependency "minitest-stub-const", "~> 0.6"
+  s.add_development_dependency 'rspec-mocks', '~> 3.11.0'
```

Comment:
Haven't looked at the in more detail but would be great if we wouldn't need to add extra dep.

Reply:
We need it. In the tests you can see I am stubbing certain methods to raise errors. There is no other way to do this in the specs b/c these error would only occur if certain infra such as Redis is down. The only alternative would be to write monkey patches in the tests, which is a far worse/messy option that using RSpec's widely-used mock library (which has minitest integration)

Reply:
I get your point, but maybe we can change [where this gem is used](https://github.com/rack/rack-attack/pull/639/files#diff-4da0c4f87f4ac598efc2c860433f206defbc573cf17f7f5f37de77faf3a54f73R21) for something similar to how [exceptions are currently being forced in tests](https://github.com/rack/rack-attack/blob/main/spec/integration/offline_spec.rb#L31), and we avoid introducing a new dependency? Or are there other places where `rspec-mocks` is needed?

Reply:
Please read my `spec/acceptance/error_handling_spec.rb`. I am using mocks to trigger various errors which do not normally occur and asserting how they are handled. `rspec-mocks` was designed exactly for this sort of use case (e.g. testing how novel errors propagate through a system.)

It is certainly possible to force certain errors by strange Redis configurations, this ultimately is limited in what types of errors we can trigger. I will be using mocks further in subsequent PRs.